sashimi-webapp/src/logic/filemanager/data/file.js

```diff
@@ -136,23 +144,45 @@ File.prototype.move = function move(destFolder) {
 };
 
 /**
+ * Make a copy of a file in the current Folder
+ *
+ * @return {Promise}
+ */
+File.prototype.copy = function copy() {
+  return storage.copyFile(this.id)
+  .then((dbCopiedFile) => {
+    const copiedFile = new File(dbCopiedFile.file_id, dbCopiedFile.file_name, dbCopiedFile.file_path, this);
+    this.parentFolder.childFileList.push(copiedFile);
+    idMap.addFileToMap(copiedFile.id, copiedFile);
+  });
+};
+
+/**
  * Rename file
  *
  * @param {String} newFileName
  * @return {Promise}
  */
 File.prototype.rename = function rename(newFileName) {
-  return new Promise((resolve, reject) => {
-    if (hasSameFileName.call(this, newFileName)) {
-      reject(ERROR_SAME_FILE_NAME);
-    }
+  if (!(this instanceof Folder)) {
+    throw new Error(ERROR_NOT_FILE_INSTANCE);
+  }
 
-    resolve();
-  })
-  .then(() => storage.renameFile(this.id, newFileName))
-  .then(() => {
-    const oldFileName = this.name;
-    this.name = newFileName;
-    this.path = this.path.replace(oldFileName, newFileName);
-  });
+  newFileName = newFileName.trim();
+  if (!newFileName) {
```

Comment:
I think even after `newFileName.trim()`, it will still return `''`. 
Might be better if you check with `newFileName.length` instead